### PR TITLE
Fixture resource directory role collection

### PR DESCRIPTION
**Pull request by rui1610**
## Purpose

* Add unit test for the directory resource

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: unit tests
```

## How to Test

* Test the code

```
go test ./...
```

## What to Check

Verify that the following are valid

* All tests should run successfully

## Other Information

Currently there is an issue to execute the unit test for the `directory` resource leading to the following error message:

`Error: API Error Deleting Resource Directory
            Received response with unexpected status [Status: 404; Correlation ID:xxxxxx`

This needs to be fixed, so that this PR can be merged. According too @lechnerc77 this is an issue based on the client that needs to be fixed by @v0lkc .

### CODE REVIEW

File: internal/provider/resource_directory_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				},
+			},
+		})
+	})
+}
+
+func hclResourceDirectoryParent(resourceName string, displayName string, description string) string {
+	return fmt.Sprintf(`resource "btp_directory" "%s" {
+        name      = "%s"
+        description    = "%s"
```
Comment (v0lkc) [inline internal/provider/resource_directory_test.go]:
```suggestion
        name        = "%s"
        description = "%s"
```

File: internal/provider/resource_directory_role_collection_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			},
+		})
+	})
+}
+
+func hclResourceDirectoryRoleCollection(resourceName string, directoryId string, displayName string, description string) string {
+	return fmt.Sprintf(`resource "btp_directory_role_collection" "%s" {
+        directory_id 		= "%s"
+        name      			= "%s"
+        description      	= "%s"
```
Comment (v0lkc) [inline internal/provider/resource_directory_role_collection_test.go]:
```suggestion
        directory_id = "%s"
        name         = "%s"
        description  = "%s"
```

File: internal/provider/resource_directory_test.go
Comment (lechnerc77) [inline internal/provider/resource_directory_test.go:1]: This file should be deleted and provided via a dedicatd PR.

File: docs/resources/directory_role_collection_assignment.md
Comment (lechnerc77) [inline docs/resources/directory_role_collection_assignment.md:1]: Taking the fixture in this PR as basis the PR is about the role colections on directory level. Then the docs around role collection assignment shouldn't be part of this PR

File: internal/provider/resource_directory_role_collection_assignment.go
Comment (lechnerc77) [inline internal/provider/resource_directory_role_collection_assignment.go:1]: Can you move this to another PR for role collection assignments?

File: internal/provider/resource_directory_role_collection_test.go
Comment (lechnerc77) [inline internal/provider/resource_directory_role_collection_test.go:1]: I would add one more test case with description and check that the role(s) are assigned i.e. that the number of rolerefernces is one in this case. 

Please also check with two roles in the collection as we had some mapping issues on the other levels with that scenario
